packages/core/src/node/light-node/index.ts

```diff
@@ -98,29 +96,26 @@ export class LightNode extends ZkopruNode {
     slasher,
     bootstrapHelper,
   }: {
-    provider: provider
+    provider: BaseProvider
     address: string
     db: DB
-    slasher?: Account
+    slasher?: Signer
     accounts?: ZkAccount[]
     bootstrapHelper: BootstrapHelper
   }): Promise<LightNode> {
-    if (!provider.connected) throw Error('provider is not connected')
+    // if (!provider.) throw Error('provider is not connected')
     if (!bootstrapHelper)
       throw Error('You need bootstrap node to run light node')
-    const web3: Web3 = new Web3(provider)
     // Add zk account to the web3 object if it exists
     const tracker = new Tracker(db)
     if (accounts) {
       await tracker.addAccounts(...accounts)
-      for (const account of accounts) {
-        web3.eth.accounts.wallet.add(account.toAddAccount())
-      }
     }
-    const l1Contract = new L1Contract(web3, address)
+    const l1Contract = new L1Contract(provider, address)
     // retrieve l2 chain from database
-    const networkId = await web3.eth.net.getId()
-    const chainId = await web3.eth.getChainId()
+    const network = await provider.getNetwork()
+    const networkId = network.chainId // todo
```

Comment:
May be we should compare latest transaction `v` field and chainId for security.
Instead of comparing `networkId` and `chainId` for preventing replay attack.